dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImplTest.java

```diff
@@ -1277,6 +1274,259 @@ public void flushVanityURLCacheRight() throws ServletException, IOException {
+            contentlet2.setProperty(publishField.variable(), afterTomorrow);
+            ContentletDataGen.checkin(contentlet2);
+
+            checkFromElasticSearch(publishField, tomorrow, afterTomorrow, contentlet1);
+
+            checkFromDataBase(publishField, tomorrow, afterTomorrow, contentlet1);
+        }finally {
+            LanguageDataGen.remove(language1);
+            LanguageDataGen.remove(language2);
+
+            FieldDataGen.remove(publishField);
+            ContentTypeDataGen.remove(contentType);
+
+            ContentletDataGen.remove(contentlet1);
+            ContentletDataGen.remove(contentlet2);
+        }
+    }
+
+    /**
+     * Method to test: {@link ESContentletAPIImpl#checkin(Contentlet, User, boolean)}
+     * When:
+     * - Set uniquePublishExpireDate to true
+     * - Create a ContentType with publish date
+     * - Create two Language
+     * - Create a {@link Contentlet} with the first language and publish date for tomorrow
+     * - Create a {@link Contentlet} with the second language and publish date for after tomorrow
+     * Should: Both {@link Contentlet} have after tomorrow as publish date
+     *
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    @Test
+    public void differentVersionWithDifferentPublishDateAndUniquePublishExpireDate() throws DotDataException, DotSecurityException {
+        final boolean uniquePublishExpireDate = ContentletTransformer.isUniquePublishExpireDatePerLanguages();
+        ContentletTransformer.setUniquePublishExpireDatePerLanguages(true);
+
+        try {
+            final Language language1 = new LanguageDataGen().nextPersisted();
+            final Language language2 = new LanguageDataGen().nextPersisted();
+
+            final Field publishField = new FieldDataGen().defaultValue(null)
+                    .type(DateTimeField.class).next();
+
+            final ContentType contentType = new ContentTypeDataGen()
+                    .field(publishField)
+                    .publishDateFieldVarName(publishField.variable())
+                    .nextPersisted();
+
+            final Calendar calendar = Calendar.getInstance();
+            calendar.add(Calendar.DATE, 1);
+            final Date tomorrow = calendar.getTime();
+
+            calendar.add(Calendar.DATE, 1);
+            final Date afterTomorrow = calendar.getTime();
+
+            final Contentlet contentlet1 = new ContentletDataGen(contentType)
+                    .setProperty(publishField.variable(), tomorrow)
+                    .languageId(language1.getId())
+                    .nextPersisted();
+
+            final Contentlet contentlet2 = ContentletDataGen.checkout(contentlet1);
+            contentlet2.setLanguageId(language2.getId());
+            contentlet2.setProperty(publishField.variable(), afterTomorrow);
+            ContentletDataGen.checkin(contentlet2);
+
+            checkFromElasticSearch(publishField, afterTomorrow, afterTomorrow, contentlet1);
```

Comment:
I don't get why the publish date for content2 is never verified

Reply:
The methods are checking both versiona